src/net_processing.cpp

```diff
@@ -3495,20 +3496,20 @@ void PeerLogicValidation::CheckForStaleTipAndEvictPeers(const Consensus::Params
 
     if (connman == nullptr) return;
 
-    int64_t time_in_seconds = GetTime();
+    const auto time = GetTime<std::chrono::seconds>();
 
-    EvictExtraOutboundPeers(time_in_seconds);
+    EvictExtraOutboundPeers(time);
 
-    if (time_in_seconds > m_stale_tip_check_time) {
+    if (time > m_stale_tip_check_time) {
         // Check whether our tip is stale, and if so, allow using an extra
         // outbound peer
         if (!fImporting && !fReindex && connman->GetNetworkActive() && connman->GetUseAddrmanOutgoing() && TipMayBeStale(consensusParams)) {
-            LogPrintf("Potential stale tip detected, will try using extra outbound peer (last tip update: %d seconds ago)\n", time_in_seconds - g_last_tip_update);
+            LogPrintf("Potential stale tip detected, will try using extra outbound peer (last tip update: %d seconds ago)\n", count_seconds(time - g_last_tip_update.load()));
```

Comment:
The type of `time - g_last_tip_update.load()` should be assumed to be `std::chrono::something`, but the exact type should be assumed to be unknown, since it can change in a future refactor. Therefore, we need to somehow specify what precision we want to log to the debug log.

Using the `count_*` helpers is one way and I bet there are also other ways to achieve the same.

Reply:
I recommend reviewing #18638 , which adds documentation to `count_seconds`